netutil/netutil.go

```diff
@@ -0,0 +1,50 @@
+package netutil
+
+import (
+	"net"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+)
+
+var (
+	getInterfaceAddrs = (*net.Interface).Addrs
+)
+
+// Parses network interfaces and returns those having an address conformant to RFC1918
+func PrivateNetworkInterfaces(logger log.Logger) []string {
+	ints, err := net.Interfaces()
+	if err != nil {
+		level.Warn(logger).Log("msg", "error getting interfaces", "err", err)
+	}
+	return privateNetworkInterfaces(ints, logger)
+}
+
+// private testable function that checks each given interface
+func privateNetworkInterfaces(all []net.Interface, logger log.Logger) []string {
+	var privInts []string
+	for _, i := range all {
+		addrs, err := getInterfaceAddrs(&i)
+		if err != nil {
+			level.Warn(logger).Log("msg", "error getting addresses", "inf", i.Name, "err", err)
+		}
+		for _, a := range addrs {
+			s := a.String()
+			ip, _, err := net.ParseCIDR(s)
+			if err != nil {
+				level.Warn(logger).Log("msg", "error getting ip address", "inf", i.Name, "addr", s, "err", err)
+			}
+			if ip.IsPrivate() {
+				level.Info(logger).Log(i.Name, "is private with address", s)
```

Comment:
Logging in a library for something low-level like this feels a little too verbose to me. Maybe `debug` for most of these and returning an error in the top-level function? That way the caller can decide how to communicate the issue to the user.

Reply:
When you would you want this info logged, as a library user?  It feels like something you would want when you were coding the library.

Reply:
Thanks for pointing this out, I meant to remove it.